chcemvediet/apps/anonymization/templatetags/chcemvediet/anonymization.py

```diff
@@ -12,29 +16,23 @@
 def anonymize(context, inforequest, content):
     request = context[u'request']
     prog = generate_user_pattern(inforequest)
-    if not inforequest.anonymized_for(request.user) or not prog.pattern:
+    if not inforequest.anonymized_for(request.user):
         return content
-    return prog.sub(ANONYMIZATION_STRING, content)
+    return anonymize_string(prog, content)
 
 @register.simple_tag(takes_context=True)
 def anonymize_html(context, inforequest, html_content):
-    """
-    Anonymize user in each tag of html_content.
-    """
     request = context[u'request']
     prog = generate_user_pattern(inforequest)
-    if not inforequest.anonymized_for(request.user) or not prog.pattern:
+    if not inforequest.anonymized_for(request.user):
         return html_content
-    root = lxml.html.fromstring(html_content)
-    for t in root.findall(u".//", {}):
-        for tt in list(t):
-            if tt.tail is None:
-                continue
-            tt.tail = prog.sub(ANONYMIZATION_STRING, tt.tail)
-        if t.text is None:
-            continue
-        t.text = prog.sub(ANONYMIZATION_STRING, t.text)
-    return lxml.html.tostring(root)
+    try:
+        return anonymize_markup(prog, html_content, etree.HTMLParser(), u'.//', {})
+    except Exception as e:
+        trace = unicode(traceback.format_exc(), u'utf-8')
+        cron_logger.error(u'anonymize_html has failed.\n An '
+                          u'unexpected error occured: {}\n{}'.format(e.__class__.__name__, trace))
+        return u'Obsah sa nepodarilo načítať.'
```

Comment:
@martinmacko47 Na jednu vec som zabudol. Nebol osetreny pripad ak parsovanie html prilohy z nejakeho dovodu padne. Preto som to zabalil do try, except. Co mam v takomto pripade vratit, aby to videl pouzivatel na obrazovke ?

Reply:
@viliambalaz Kludne nateraz zobrazit takto error hlasku, ako si spravil. Akurat ju bude treba lokalizovat. Tj. ze v kode nebude text, ale iba formalne id textu a samotny text bude v prekladovom subore.

Reply:
:ok: